.circleci/config.yml

```diff
@@ -0,0 +1,74 @@
+version: 2
+jobs:
+  test:
+    docker:
+      - image: ruby:2.5.0-alpine
+        environment:
+          BUNDLE_RETRY: 3
+          BUNDLE_PATH: vendor/bundle
+    working_directory: ~/rds-scheduler/src
+    steps:
+      - checkout:
+          path: ~/rds-scheduler
+      - run:
+          name: Install build dependencies
+          command: apk add -U g++ make
+      - run:
+          name: Install ruby dependencies
+          command: bundle install --gemfile=Gemfile --path=vendor/bundle
+      - run:
+          name: Run code analyser
+          command: bundle exec rubocop
+      - run:
+          name: Run RSpec
+          command: bundle exec rspec spec/
+  build:
+    machine: true
+    steps:
+      - checkout
+      - run:
+          name: Build Image
+          command: docker build -t appvia/rds-scheduler:$CIRCLE_SHA1 .
+  push_latest:
+    machine: true
+    steps:
+      - checkout
+      - run:
+          name: Push to Quay with tag latest
+          command: |
+            docker build -t quay.io/appvia/rds-scheduler:latest .
```

Comment:
Is it possible to reuse the built image from the `build` step above? If not, any point in having these two separate steps?

Reply:
I actually had problems getting this to work last time I tried, when splitting the tasks across jobs. So the plan was:
- Build image: Occur on every push / tag
- Push latest/tag: Re-use the image from previous step, only occurs on certain events (as defined in the workflow)

But I couldn't get the docker context persisting across jobs. Tested this a while ago but will try again (probably in a follow-up PR).

Reply:
Fair enough 👍